spec/values/calculation/abs.hrx

```diff
@@ -0,0 +1,255 @@
+terseness' sake isn't tested explicitly.
+
+<===>
+================================================================================
+<===> zero/input.scss
+a {b: abs(0)}
+
+<===> zero/output.css
+a {
+  b: 0;
+}
+
+<===>
+================================================================================
+<===> positive/integer/input.scss
+a {b: abs(1)}
+
+<===> positive/integer/output.css
+a {
+  b: 1;
+}
+
+<===>
+================================================================================
+<===> negative/decimal/input.scss
+a {b: abs(-5.6px)}
+
+<===> negative/decimal/output.css
+a {
+  b: 5.6px;
+}
+
+<===>
+================================================================================
+<===> operation/variable/input.scss
+a {
+  --test: 5;
+  b: abs(1px + 2px - var(--test))
+}
+
+<===> operation/variable/output.css
+a {
+  --test: 5;
+  b: abs(3px - var(--test));
+}
+
+<===>
+================================================================================
+<===> operation/preserves_units/input.scss
+a {b: abs(-7px / 4em) * 1em}
+
+<===> operation/preserves_units/output.css
+a {
+  b: 1.75px;
+}
+
+<===>
+================================================================================
+<===> operation/incompatible_units/input.scss
+a {b: abs(17px * 5%)}
+
+<===> operation/incompatible_units/error
+Error: 85px*% isn't a valid CSS value.
+  ,
+1 | a {b: abs(17px * 5%)}
+  |       ^^^^^^^^^^^^^^
+  '
+  input.scss 1:7  root stylesheet
+
+<===>
+================================================================================
+<===> operation/integer/input.scss
+a {b: abs(17 * 5%)}
+
+<===> operation/integer/output.css
+a {
+  b: 85%;
+}
+
+<===> operation/integer/warning
+DEPRECATION WARNING: Passing percentage units to the global abs() function is deprecated.
+In the future, this will emit a CSS abs() function to be resolved by the browser.
+To preserve current behavior: math.abs(85%)
+To emit a CSS abs() now: abs(#{85%})
+More info: https://sass-lang.com/d/abs-percent
+  ,
+1 | a {b: abs(17 * 5%)}
+  | ^^^^^^^^^^^^^^^^^^^
+  '
+    input.scss 1:1  root stylesheet
+
+<===>
+================================================================================
+<===> preserved/operation/percentage/input.scss
+a {b: abs(17px + 5%)}
+
+<===> preserved/operation/percentage/output.css
+a {
+  b: abs(17px + 5%);
+}
+
+<===>
+================================================================================
+<===> preserved/variable/input.scss
+a {
+  --test: 5;
+  b: abs(var(--test))
+}
+
+<===> preserved/variable/output.css
+a {
+  --test: 5;
+  b: abs(var(--test));
+}
+
+<===>
+================================================================================
+<===> preserved/unquoted_string/input.scss
+a {
+  b: abs(test);
+}
+
+<===> preserved/unquoted_string/output.css
+a {
+  b: abs(test);
+}
+
+<===>
+================================================================================
+<===> named_argument/input.scss
+a {b: abs($number: -3)}
```

Comment:
It's probably worth another test to verify that there's a strict delineation between "parsing `abs()` as a Sass function" and "parsing `abs()` as a calculation". For example, `abs($number: var(--c))` should fail.